app/Http/Controllers/Api/ApiCsvParserController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Services\CsvParserService;
+use Illuminate\Http\Request;
+use function dd;
+use function storage_path;
+
+class ApiCsvParserController extends Controller
+{
+    /** @var CsvParserService */
+    private CsvParserService $csvParserService;
+
+    public function __construct(CsvParserService $csvParserService)
+    {
+        $this->csvParserService = $csvParserService;
+    }
```

Comment:
https://php.watch/versions/8.0/constructor-property-promotion
Could be refactored as 

```suggestion
    public function __construct(private CsvParserService $csvParserService)
    {
    }
```